modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/helper/DDMFormEvaluatorHelper.java

```diff
@@ -172,6 +172,32 @@ public DDMFormEvaluatorEvaluateResponse evaluate() {
 		return formEvaluatorEvaluateResponse.build();
 	}
 
+	protected Map<String, Object> calculateAction(DDMFormRule ddmFormRule) {
```

Comment:
Here I'm returning the fields values evaluated by some rule. To do that, I had to use the function that literally calculates the fields. And as this function updates the __ddmFormFieldsPropertyChanges_, I'm cleaning the object after invokes this evaluation. Before that, I'm running through the object to populate a auxiliar object that I'm returning in this function to the _containsCalculatedAction_ get all the necessary info.